subgraphs/inverse-finance/src/common/utils.ts

```diff
@@ -0,0 +1,39 @@
+import { BigDecimal, BigInt } from "@graphprotocol/graph-ts";
+import { SECONDS_PER_HOUR, SECONDS_PER_DAY } from "./constants";
+
+// Converts snake case to kebab case and appends a hyphen.
+// (e.g. "TRADING_FEE" to "trading-fee-"), mainly used to create entity IDs
+export function enumToPrefix(snake: string): string {
+  return snake.replace("_", "-") + "-";
+}
+
+// Prefix an ID with a enum string in order to differentiate IDs
+// e.g. combine TRADING_FEE and 0x1234 into TRADING-FEE-0x1234
+export function prefixID(ID: string, enumString1: string, enumString2: string | null = null): string {
+  let prefix = enumToPrefix(enumString1);
+  if (enumString2 != null) {
+    prefix += enumToPrefix(enumString2!);
+  }
+  return prefix + ID;
+}
+
+// returns 10^decimals
+export function decimalsToBigDecimal(decimals: i32): BigDecimal {
+  let bd = BigDecimal.fromString("1");
+  for (let i = 0; i < decimals; i++) {
+    bd = bd.times(BigDecimal.fromString("10"));
+  }
+  return bd;
+}
+
+// returns hours (24-hour format) from timestamp
+export function timestampToHH(timestamp: i64): BigInt {
+  let secondsPastMidnight = timestamp % SECONDS_PER_DAY;
+  let hours = secondsPastMidnight / SECONDS_PER_HOUR;
+  return BigInt.fromI32(hours);
+}
+
+export function BigDecimalTruncateToBigInt(x: BigDecimal): BigInt {
```

Comment:
Doesn't `BigDecimal` already have a `truncate` function?

Reply:
Frankly, I am confused by what the `truncate` function does - it seems to truncate digits left of zero? In addition, it returns another BigDecimal. In my case, I need to convert the result of normalized daily emission amount (a BigDecimal because of division) to BigInt.

Reply:
Yes it's not well-documented. You can refer to: https://github.com/0xbe1/compound-v2-subgraph/blob/master/tests/example.test.ts

Reply:
The test is helpful! To use `BigDecimal.truncate` in my case, I will still need to detect number of decimal places in my BigDecimal, call truncate, convert to string, then to BigInt. I found it is more convenient to use my own function